File: frontend/src/models/Student.ts
import IFaculty from "./Faculty";
import IProgram from "./Program";
import IStatus from "./Status";

interface IStudent {
    studentId: number;
    fullName: string;
    dateOfBirth: string;
    gender: string;
    facultyId: number;
    batch: string;
    programId: number;
    address: string;
    email: string;
    phoneNumber: string;
    statusId: number;
    faculty: IFaculty;
    program: IProgram;
    status: IStatus;
    createdAt: string;
}

// - Mã số sinh viên
// - Họ tên
// - Ngày tháng năm sinh
// - Giới tính
// - Khoa 
// - Khóa 
// - Chương trình 
// - Địa chỉ
// - Email
// - Số điện thoại liên hệ
// - Tình trạng sinh viên

export default IStudent;

File: backend/Entities/Student.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace backend.Entities
{
    public class Student
    {
        public int StudentId { get; set; }  // Mã số sinh viên
        public string FullName { get; set; }   // Họ tên
        public DateTime DateOfBirth { get; set; }  // Ngày tháng năm sinh 
        public string Gender { get; set; }     // Giới tính
        public string Batch { get; set; }      // Khóa
        public string Address { get; set; }    // Địa chỉ
        public string Email { get; set; }      // Email
        public string PhoneNumber { get; set; }  // Số điện thoại liên hệ
        public int StatusId { get; set; }
        public int ProgramId { get; set; }
        public int FacultyId { get; set; }
        
        // Navigation properties
        public Status Status { get; set; }     // Tình trạng sinh viên
        public StudyProgram Program { get; set; }    // Chương trình
        public Faculty Faculty { get; set; }
        public List<StudentNotification> SubscribeToNotifications { get; set; } = new List<StudentNotification>();
        public DateTime CreatedAt { get; set; }
    }
}

